debian/snapd.prerm

```diff
@@ -0,0 +1,16 @@
+#! /bin/sh
+# prerm script for snapd
+
+set -e
+
+if [ "$1" = "remove" -o "$1" = "upgrade" ]; then
```

Comment:
why do you not want to try to stop services unless removing or upgrading? the other ways the prerm script is called is when you most need help. Wrap each one in `systemctl is-active --quiet $UNIT`.